packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -237,51 +245,39 @@ class RenderableTiledMap {
 
 abstract class _RenderableLayer<T extends Layer> {
   final T layer;
+  final Vector2 destTileSize;
+  final TiledMap map;
 
   /// The parent [Group] layer (if it exists)
   final _RenderableGroupLayer? parent;
 
-  _RenderableLayer(this.layer, this.parent);
+  _RenderableLayer(
+    this.layer,
+    this.parent,
+    this.map,
+    this.destTileSize,
+  );
 
   bool get visible => layer.visible;
 
   void render(Canvas canvas, Camera? camera);
 
-  void handleResize(Vector2 canvasSize) {}
+  void handleResize(Vector2 canvasSize);
 
-  void refreshCache() {}
+  void refreshCache();
 
-  double get offsetX {
-    return layer.offsetX + (parent?.offsetX ?? 0);
-  }
+  double get scaleX => destTileSize.x / map.tileWidth;
+  double get scaleY => destTileSize.y / map.tileHeight;
 
-  double get offsetY {
-    return layer.offsetY + (parent?.offsetY ?? 0);
-  }
+  late double offsetX = layer.offsetX * scaleX + (parent?.offsetX ?? 0);
 
-  double get opacity {
-    if (parent != null) {
-      return parent!.opacity * layer.opacity;
-    } else {
-      return layer.opacity;
-    }
-  }
+  late double offsetY = layer.offsetY * scaleY + (parent?.offsetY ?? 0);
 
-  double get parallaxX {
-    if (parent != null) {
-      return parent!.parallaxX * layer.parallaxX;
-    } else {
-      return layer.parallaxX;
-    }
-  }
+  late double opacity = layer.opacity * (parent?.opacity ?? 1);
 
-  double get parallaxY {
-    if (parent != null) {
-      return parent!.parallaxY * layer.parallaxY;
-    } else {
-      return layer.parallaxY;
-    }
-  }
+  late double parallaxX = layer.parallaxX * (parent?.parallaxX ?? 1);
+
+  late double parallaxY = layer.parallaxY * (parent?.parallaxY ?? 1);
```

Comment:
Any specific reason to not have these as getters?

Reply:
Which ones? Offset, Opacity, or Parallax? 

Main reason: performance.  As getters, you're calculating this every single render loop. As late-calculations, they are calculated once and cached.

Reply:
As per our discussion on discord, this is few as these values never change.